Resources/ProxyProvider/PUSPProxyProvider.py

```diff
@@ -18,49 +18,35 @@ def __init__(self, parameters=None):
 
     super(PUSPProxyProvider, self).__init__(parameters)
 
-  def getProxy(self, userDict):
-    """ Generate user proxy
+  def checkStatus(self, userDN):
+    """ Read ready to work status of proxy provider
 
-        :param dict userDict: user description dictionary with possible fields:
-               FullName, UserName, DN, Email, DiracGroup
+        :param str userDN: user DN
 
-        :return: S_OK(basestring)/S_ERROR() -- basestring is a proxy string
+        :return: S_OK(dict)/S_ERROR() -- dictionary contain fields:
+                  - 'Status' with ready to work status[ready, needToAuth]
     """
+    if not userDN.split(":")[-1]:
+      return S_ERROR('Can not found user label for DN: %s' % userDN)
+    if not self.parameters.get('ServiceURL'):
+      return S_ERROR('Can not determine PUSP service URL')
+
+    return S_OK({'Status': 'ready'})
```

Comment:
Here, and in the other `checkStatus` methods, you can simply return `S_OK/S_ERROR`. Returning always `S_OK({'Status': 'ready'})` does not add anything (and I see it is never checked).

Reply:
The dictionary has returned to be able to add more status information if the status is not `ready`.

Reply:
Can't you just return `S_OK('ready')`? Or even only `S_OK()` ?

Reply:
In this case, I am unable to add status information, such as some message or authZ URL to update the OAuth session if we will use RCAuth CA via EGI AAI.